File: app/src/main/java/net/afterday/compas/core/Game.java
package net.afterday.compas.core;

import net.afterday.compas.core.gameState.Frame;
import net.afterday.compas.core.influences.InfluencesPack;
import net.afterday.compas.core.inventory.Inventory;
import net.afterday.compas.core.inventory.items.Item;
import net.afterday.compas.core.player.Player;

/**
 * Created by Justas Spakauskas on 2/3/2018.
 */

public interface Game
{
    Frame start();
    ////////////////////////////////////
    Frame acceptInfluences(InfluencesPack influences);
    Player getPlayer();
    Inventory getInventory();
    boolean acceptCode(String code);
    Frame useItem(Item item);
}

File: engine/src/main/kotlin/net/afterday/compas/engine/core/Game.kt
package net.afterday.compas.engine.core

import net.afterday.compas.engine.core.gameState.Frame
import net.afterday.compas.engine.core.influences.InfluencesPack
import net.afterday.compas.engine.core.inventory.Inventory
import net.afterday.compas.engine.core.inventory.items.Item
import net.afterday.compas.engine.core.player.Player

interface Game {
    fun start(): Frame
    fun acceptInfluences(influences: InfluencesPack): Frame

    val player: Player
    val inventory: Inventory

    fun acceptCode(code: String): Boolean
    fun useItem(item: Item): Frame
}